File: app/src/main/java/com/martinrevert/latorrentola/database/Converters.java
package com.martinrevert.latorrentola.database;

import androidx.room.TypeConverter;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import com.martinrevert.latorrentola.model.YTS.Torrent;


import java.lang.reflect.Type;
import java.util.Date;
import java.util.List;

/**
 * Created by martin on 07/12/17.
 */

public class Converters {
    @TypeConverter
    public static List<Torrent> fromTorrentString(String value) {
        Type listType = new TypeToken<List<Torrent>>() {}.getType();
        return new Gson().fromJson(value, listType);
    }

    @TypeConverter
    public static String fromTorrentList(List<Torrent> list) {
        Gson gson = new Gson();
        String json = gson.toJson(list);
        return json;
    }

    @TypeConverter
    public static List<String> fromString(String value) {
        Type listType = new TypeToken<List<String>>() {}.getType();
        return new Gson().fromJson(value, listType);
    }

    @TypeConverter
    public static String fromArrayList(List<String> list) {
        Gson gson = new Gson();
        String json = gson.toJson(list);
        return json;
    }

    @TypeConverter
    public Date fromTimestamp(Long value) {
        return value == null ? null : new Date(value);
    }

    @TypeConverter
    public Long dateToTimestamp(Date date) {
        return date == null ? null : date.getTime();
    }




}

File: app/src/main/java/com/martinrevert/latorrentola/database/Converters.kt
package com.martinrevert.latorrentola.database

import androidx.room.TypeConverter
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import com.martinrevert.latorrentola.model.YTS.Cast
import com.martinrevert.latorrentola.model.YTS.Torrent
import java.util.Date

class Converters {
    @TypeConverter
    fun fromTorrentString(value: String?): List<Torrent>? {
        val listType = object : TypeToken<List<Torrent>>() {}.type
        return Gson().fromJson(value, listType)
    }

    @TypeConverter
    fun fromTorrentList(list: List<Torrent>?): String? {
        return Gson().toJson(list)
    }

    @TypeConverter
    fun fromString(value: String?): List<String>? {
        val listType = object : TypeToken<List<String>>() {}.type
        return Gson().fromJson(value, listType)
    }

    @TypeConverter
    fun fromArrayList(list: List<String>?): String? {
        return Gson().toJson(list)
    }

    @TypeConverter
    fun fromCastString(value: String?): List<Cast>? {
        val listType = object : TypeToken<List<Cast>>() {}.type
        return Gson().fromJson(value, listType)
    }

    @TypeConverter
    fun fromCastList(list: List<Cast>?): String? {
        return Gson().toJson(list)
    }

    @TypeConverter
    fun fromTimestamp(value: Long?): Date? {
        return value?.let { Date(it) }
    }

    @TypeConverter
    fun dateToTimestamp(date: Date?): Long? {
        return date?.time
    }
}
